### PR TITLE
RUBY-3341 Document error handling in transactions

### PR DESCRIPTION
**Pull request by comandeo**

### CODE REVIEW

File: docs/reference/transactions.txt [hunk 1]
```diff
@@ -145,6 +166,14 @@ session if one is in progress:
   # ok
   c2.database.drop
 
+Handling Errors
+---------------
+
+If a command inside the transaction fails with, the transaction may be aborted
```
Comment (alexbevi) [inline docs/reference/transactions.txt:172]: "If a command inside the transaction fails with," .... fails with what?

File: docs/reference/transactions.txt [hunk 2]
```diff
@@ -145,6 +166,14 @@ session if one is in progress:
...
+on the server. Errors that abort transactions do not have
+``TransientTransactionError`` in their error labels. An attempt to commit such
+transaction will be rejected with ``NoSuchTransaction`` error.
```
Comment (alexbevi) [inline docs/reference/transactions.txt:175]:
```suggestion
``TransientTransactionError`` in their error labels. An attempt to commit such a
transaction will be rejected with ``NoSuchTransaction`` error.
```

**Review by alexbevi — APPROVED** [review]
minor change requested, but overall LGTM

File: docs/reference/transactions.txt (hunk 1 above)
Comment (comandeo) [reply docs/reference/transactions.txt]: Fixed, thanks!

**Review by jamis — APPROVED** [review]

**Review by alexbevi — APPROVED** [review]

**Review by jamis — APPROVED** [review]

**Pull request merged**